narwhals/_arrow/dataframe.py

```diff
@@ -285,6 +287,10 @@ def schema(self: Self) -> dict[str, DType]:
     def collect_schema(self: Self) -> dict[str, DType]:
         return self.schema
 
+    def estimated_size(self: Self, unit: SizeUnit = "b") -> int | float:
```

Comment:
We are trying to avoid having defaults for compliant object methods

```suggestion
    def estimated_size(self: Self, unit: SizeUnit) -> int | float:
```